pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -367,11 +370,33 @@ public SegmentPartitionConfig getSegmentPartitionConfig() {
   }
 
   public long getMinTime() {
-    return _minTime;
+    Long minTime = extractTimeValue(_minValueMap.get(_timeColumnName));
+    if (minTime != null) {
+      return minTime;
+    }
+    return Long.MAX_VALUE;
   }
 
   public long getMaxTime() {
-    return _maxTime;
+    Long maxTime = extractTimeValue(_maxValueMap.get(_timeColumnName));
+    if (maxTime != null) {
+      return maxTime;
+    }
+    return Long.MIN_VALUE;
+  }
+
+  private Long extractTimeValue(Comparable time) {
+    if (time != null) {
+      if (time instanceof Number) {
+        return ((Number) time).longValue();
+      } else {
+        String stringValue = time.toString();
```

Comment:
why are we assuming time is always long?

Reply:
This logic is only used for Kafka HLC. It assumes time field  is always some epoch value. Maybe we should mark this as Deprecated?